cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -251,6 +251,8 @@ Future<Void> reconcileCas(Clock clock) {
                         } else if (secretName.equals(KafkaResources.kafkaSecretName(reconciliation.name()))) {
                             clusterCaSecrets.add(secret);
                             clientsCaSecrets.add(secret);
+                        } else if (secretName.equals(KafkaResources.clusterOperatorCertsSecretName(reconciliation.name()))) {
+                            // The CO certificate is excluded as it is renewed in a separate cycle
                         } else {
```

Comment:
Has this behaviour happened due to the change from this refactor? https://github.com/strimzi/strimzi-kafka-operator/pull/10023 What secret are we now catching in the `else` here? Could we not remove the `else` clause instead so we explicitly add the Secrets we care about?

Reply:
> Has this behaviour happened due to the change from this refactor? https://github.com/strimzi/strimzi-kafka-operator/pull/10023

No, it is not related to #10023. From my point of view, it originates from #6180 (that said, IIRC it did not worked at all before #6180, so it is not as if #6180 broke this)

> What secret are we now catching in the else here? Could we not remove the else clause instead so we explicitly add the Secrets we care about?

Everything not listed above it. KE, CC, ETO, EUO, ZOO. I do not think we want to have a dedicated check for each of them.

Reply:
Thanks, makes sense